src/complex-list-field/complex-list-field.component.ts

```diff
@@ -151,23 +149,56 @@ export class ComplexListFieldComponent extends AbstractListFieldComponent implem
 
   onPageChange(page: number) {
     this.currentPage = page;
-    this.paginatedIndices = this.getIndicesForPage(page);
+    this.paginatedItems = this.getItemsForPage(page);
   }
 
-  getIndicesForPage(page: number): Array<number> {
-    let start = (page - 1) * this.navigator.itemsPerPage;
-    let indices: Array<number> = Array.apply(0, Array(this.navigator.itemsPerPage))
-      .map((el, index) => start + index);
-    // check if the indices includes some numbers that are out of values index range.
-    let lastIndexDiff = indices[indices.length - 1] - (this.values.size - 1);
-    if (lastIndexDiff > 0) {
-      indices.splice(indices.length - lastIndexDiff);
-    }
-    return indices;
-  }
+  getItemsForPage(page: number): Array<PaginatedItem> {
+    let items: Array<PaginatedItem> = [];
+    let indices: Array<number>;
```

Comment:
you don't have to set the type, just move below line here and it will be OK. (init during definition)